parasolr/django/views.py

```diff
@@ -0,0 +1,73 @@
+import calendar
+import logging
+
+from django.utils.cache import get_conditional_response
+from django.views.generic.base import View
+
+from parasolr.django import SolrQuerySet
+from parasolr.solr import SolrClientException
+from parasolr.utils import solr_timestamp_to_datetime
+
+
+logger = logging.getLogger(__name__)
+
+
+class SolrLastModifiedMixin(View):
+    """View mixin to add last modified headers based on Solr.
+    By default, searches entire solr collection and returns the most
+    recent last modified value (assumes **last_modified** field).
+    To filter for items specific to your view, either
+    set :attr:`solr_lastmodified_filters` or
+    implement :meth:`get_solr_lastmodified_filters`.
+    """
+
+    #: solr query filter for getting last modified date
+    solr_lastmodified_filters = {}   # by default, find all
+
+    def get_solr_lastmodified_filters(self):
+        '''Get filters for last modified Solr query. By default returns
+        :attr:`solr_lastmodified_filters`.'''
+        return self.solr_lastmodified_filters
+
+    def last_modified(self):
+        '''Return last modified :class:`datetime.datetime` from the
+        specified Solr query'''
+        filter_qs = self.get_solr_lastmodified_filters()
+        sqs = SolrQuerySet().filter(**filter_qs) \
+            .order_by('-last_modified').only('last_modified')
```

Comment:
is there any way to hint to solr here that we only want one result? not sure if there's a performance benefit to be had but just wondering.

Reply:
the `sqs[0]` notation a few lines down does exactly that (I just checked the logs to confirm that I'm not making this up 🙂  )

Reply:
holy cow, that's cool.